docs/app-modules/interchain-accounts/transactions.md

```diff
@@ -0,0 +1,48 @@
+<!--
+order: 5
+-->
+
+# Transactions
+
+Learn about Interchain Accounts transaction execution {synopsis}
+
+## Executing a transaction
+
+As described in [Authentication Modules](./auth-modules.md#trysendtx) transactions are executed using the interchain accounts controller API and require a `Base Application` as outlined in [ICS30 IBC Middleware](https://github.com/cosmos/ibc/tree/master/spec/app/ics-030-middleware) to facilitate authentication. The method of authentication remains unspecified to provide flexibility for the authentication module developer.
+
+Transactions are executed via the ICS27 [`TrySendTx` API](./auth-modules.md#trysendtx). This must be invoked through the Interchain Accounts authentication module, and follows the outlined path of execution below. Packet relaying semantics provided by the IBC core transport, authentication and ordering (IBC/TAO) layer are omitted for brevity.
+
+![send-tx-flow](../../assets/send-interchain-tx.png "Transaction Execution")
+
+## Atomicity
+
+As the Interchain Accounts module supports the execution of multiple transactions using the Cosmos SDK `Msg` interface, it provides the same atomicity guarantees as Cosmos SDK-based applications, leveraging the [`CacheMultiStore`](https://docs.cosmos.network/master/core/store.html#cachemultistore) architecture provided by the `Context` type. 
+
+When a host chain receives an Interchain Accounts packet and successfully deserializes its message content, each `Msg` is authenticated and executed using a cached storage object. A new `CacheMultiStore` is obtained via the Cosmos SDK `ctx.CacheContext()` method. State changes are then performed within the context of a branched key-value store and commited when `writeCache()` is invoked. This storage mechanism ensures that state changes are committed if and only if all transactions succeed.
```

Comment:
```suggestion
```

Reply:
I don't think we should go into this level of detail, personally. I think we can probably remove the `executeTx` definition below also, or maybe just link to the code.

Reply:
My reasoning for this is that I feel we shouldn't overcomplicate things for developers.

Reply:
We should add this into the godoc of executeTx. Agree with keeping these docs simple. I'd just say that the transaction is atomic and the state changes are only written if all the messages succeed

Reply:
Okay, we can remove this section. We already have something similar as inline comments within `executeTx`: 

```
// CacheContext returns a new context with the multi-store branched into a cached storage object
// writeCache is called only if all msgs succeed, performing state transitions atomically
```

I essentially had just paraphrased the above to flesh out the docs content some more as I wasn't sure if it would be too sparse without.